contrib/docker-compose/standalone/haproxy/etc/haproxy-internal.cfg

```diff
@@ -0,0 +1,56 @@
+global
+    log stdout format raw local0
+
+    # Allow local admin socket
+    stats socket /usr/local/etc/haproxy/haproxy.admin.sock mode 660 level admin expose-fd listeners
+    stats timeout 30s
+
+defaults
+    log global
+
+    option dontlognull
+    option http-server-close
+    option splice-response
+    option clitcpka
+    option srvtcpka
+    option tcp-smart-accept
+    option tcp-smart-connect
+    option contstats
+    retries 3
+
+    timeout http-request    5s
+    timeout http-keep-alive 5s
+    timeout connect         5s
+    timeout client          60s
+    timeout client-fin      60s
+    timeout tunnel          40m         # timeout to use with WebSocket and CONNECT
+    timeout server          150s
+    timeout tarpit          15s
+    timeout queue           10s
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Peerings
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+peers haproxy-peers
+    peer "${HOSTNAME}"      "${HOST_IP}":1395
+    peer haproxy            haproxy:1395
```

Comment:
Name resolution in docker-compose is racy - if A depends on B, but B has a reference to the name of A in its config, sometimes B will start up before A exists, and the name of A won't resolve. This is currently an issue in the prometheus config, but in that case we can just restart the prometheus container and no harm done. Here it's a little more critical...

Reply:
Indeed, it's a known "issue" with docker-compose, but restart will help. Operators must keep an eye unfortunately.